src/main/java/com/google/cloud/tools/maven/CloudSdkDownloader.java

```diff
@@ -0,0 +1,93 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.maven;
+
+import com.google.cloud.tools.managedcloudsdk.BadCloudSdkVersionException;
+import com.google.cloud.tools.managedcloudsdk.ConsoleListener;
+import com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVerificationException;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVersionMismatchException;
+import com.google.cloud.tools.managedcloudsdk.ProgressListener;
+import com.google.cloud.tools.managedcloudsdk.UnsupportedOsException;
+import com.google.cloud.tools.managedcloudsdk.command.CommandExecutionException;
+import com.google.cloud.tools.managedcloudsdk.command.CommandExitException;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import com.google.cloud.tools.managedcloudsdk.install.SdkInstallerException;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.nio.file.Path;
+
+public class CloudSdkDownloader {
+
+  private ManagedCloudSdk managedCloudSdk;
+  private CloudSdkMojo mojo;
+
+  CloudSdkDownloader(CloudSdkMojo mojo) {
+    this.mojo = mojo;
+    try {
+      this.managedCloudSdk = new ManagedCloudSdkFactory(mojo.getCloudSdkVersion()).newManagedSdk();
+    } catch (UnsupportedOsException | BadCloudSdkVersionException ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  @VisibleForTesting
+  public void setManagedCloudSdk(ManagedCloudSdk managedCloudSdk) {
+    this.managedCloudSdk = managedCloudSdk;
+  }
+
+  /**
+   * Downloads/installs/updates the Cloud SDK
+   *
+   * @return The cloud SDK installation directory
+   */
+  public Path downloadCloudSdk() {
+    try {
+      ProgressListener progressListener = new NoOpProgressListener();
+      ConsoleListener consoleListener = new CloudSdkDownloaderConsoleListener(mojo.getLog());
+
+      if (!managedCloudSdk.isInstalled()) {
+        managedCloudSdk.newInstaller().install(progressListener, consoleListener);
+      }
+
+      if (!managedCloudSdk.hasComponent(SdkComponent.APP_ENGINE_JAVA)) {
+        managedCloudSdk
+            .newComponentInstaller()
+            .installComponent(SdkComponent.APP_ENGINE_JAVA, progressListener, consoleListener);
+      }
+
+      if (!managedCloudSdk.isUpToDate()) {
+        managedCloudSdk.newUpdater().update(progressListener, consoleListener);
+      }
+
+      return managedCloudSdk.getSdkHome();
+    } catch (IOException
+        | SdkInstallerException
+        | ManagedSdkVersionMismatchException
+        | InterruptedException
+        | CommandExecutionException
+        | CommandExitException
+        | ManagedSdkVerificationException ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  /** Verifies the cloud sdk installation */
+  public void checkCloudSdk() {
```

Comment:
Check doesn't seem to fit into this class as the `downloader`. Perhaps a separate class for check?